### PR TITLE
Update .gitignore with other Haskell build artifacts

### PR DESCRIPTION
**Pull request by RyanGlScott**
I took this list from https://github.com/github/gitignore/blob/b34e5b4a7c6a620f71aa7042a9f408c682a68426/Haskell.gitignore.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Ping?

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,22 @@
 .ghc.environment.*
 /dist-newstyle/
 /dist/
+cabal-dev
```
Comment (hvr) [inline .gitignore]: is this a file or a folder?

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,22 @@
...
+*.o
+*.hi
+*.chi
+*.chs.h
+*.dyn_o
+*.dyn_hi
+.hpc
+.hsenv
```
Comment (hvr) [inline .gitignore]: tool/program emits this? is this a folder or a file?

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,22 @@
...
+.cabal-sandbox/
```
Comment (hvr) [inline .gitignore]: I'm missing a `/` (like for `/dist/` and `/dist-newstyle/`)

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,22 @@
...
+cabal.sandbox.config
```
Comment (hvr) [inline .gitignore]: leading `/`

File: .gitignore [hunk 5]
```diff
@@ -2,3 +2,22 @@
...
+*.prof
+*.aux
+*.hp
+*.eventlog
+.stack-work/
+cabal.project.local
+cabal.project.local~
```
Comment (hvr) [inline .gitignore]: redundant

File: .gitignore [hunk 6]
```diff
@@ -2,3 +2,22 @@
...
+.hpc
+.hsenv
+.cabal-sandbox/
+cabal.sandbox.config
+*.prof
+*.aux
+*.hp
+*.eventlog
+.stack-work/
+cabal.project.local
```
Comment (hvr) [inline .gitignore]: leading `/`

File: .gitignore [hunk 7]
```diff
@@ -2,3 +2,22 @@
...
+*.dyn_hi
+.hpc
+.hsenv
+.cabal-sandbox/
+cabal.sandbox.config
+*.prof
+*.aux
+*.hp
+*.eventlog
+.stack-work/
```
Comment (hvr) [inline .gitignore]: leading `/`

File: .gitignore [hunk 8]
```diff
@@ -2,3 +2,22 @@
...
+cabal.project.local
+cabal.project.local~
+.HTF/
```
Comment (hvr) [inline .gitignore]: file or folder?

**Comment by RyanGlScott** [comment]
Keep in mind that this is taken from GitHub's standard Haskell `.gitignore` file [here](https://github.com/github/gitignore/blob/b34e5b4a7c6a620f71aa7042a9f408c682a68426/Haskell.gitignore), so if you want to see these settings change, I'd recommend making a pull request there. (As a result, I can't answer any of these "file or folder?" questions, since I'm not the one who added these.)

**Comment by hvr** [comment]
Alright, let's merge this... it's still not perfect but it's an improvement :-)

**Pull request merged**